filesystem/mountpoint.go

```diff
@@ -240,6 +241,14 @@ func findMainMount(filesystemMounts []*Mount) *Mount {
 		uncontainedSubtrees := make(map[string]bool)
 		addUncontainedSubtreesRecursive(uncontainedSubtrees, mntNode, allUncontainedSubtrees)
 		if len(uncontainedSubtrees) != len(allUncontainedSubtrees) {
+			if mnt.Subtree == "/"+baseDirName && !allSubtrees["/"] {
+				metadataPath = mnt.Path
+			} else if len(lookuppath) > 0 &&
+				(mainMount == nil || mainMount.ReadOnly ||
+					(strings.HasPrefix(lookuppath, mnt.Path) &&
```

Comment:
It's tough to review this because the logic of this function is already pretty complex, in order to handle non-root subtrees.  How about separating out looking for a `/` or `/.fscrypt` subtree into a separate function?  So something like:

```
func findMainMount(filesystemMounts []*Mount) *Mount {                           
        mnt := findRootMainMount(filesystemMounts)                               
        if mnt != nil {                                                          
                return mnt                                                       
        }                                                                        
        return findNonRootMainMount(filesystemMounts)                            
}                                                                                
```

`findRootMainMount()` would be new, while `findNonRootMainMount()` would be the existing `findMainMount()`.